src/convert.rs

```diff
@@ -1115,6 +1116,60 @@ pub fn elf_to_tbf(
         }
     }
 
+    if ecdsa_nist_p256_private_key.is_some() {
+        let private_key_path_str = ecdsa_nist_p256_private_key.unwrap();
+        let private_key_path = Path::new(&private_key_path_str);
+        let private_key_contents = read_rsa_file(private_key_path).unwrap_or_else(|e| {
+            panic!(
+                "Failed to read private key from {:?}: {:?}",
+                private_key_path, e
+            );
+        });
+
+        let rng = rand::SystemRandom::new();
+
+        let key_pair = ring::signature::EcdsaKeyPair::from_pkcs8(
+            &ring::signature::ECDSA_P256_SHA256_FIXED_SIGNING,
+            &private_key_contents,
+        )
+        .unwrap_or_else(|e| {
+            panic!("ECDSA NIST P256 could not be parsed: {:?}", e);
+        });
+
+        let signature = key_pair
+            .sign(&rng, &output[0..tbfheader.binary_end_offset() as usize])
+            .map_err(|e| {
+                panic!("Could not generate ECDSA NIST P256 signature: {:?}", e);
+            })
+            .unwrap();
+
+        let sig_len = signature.as_ref().len();
+        let nist_p256_len = mem::size_of::<header::TbfHeaderTlv>()
+            + mem::size_of::<header::TbfFooterCredentialsType>()
+            + sig_len; // Signature + key is 32 bytes long
```

Comment:
Is the key stored? And is the signature 32 bytes or 64?

Reply:
The comment is just out of date, I have removed it.

The key isn't stored and the signature is 64-bytes long